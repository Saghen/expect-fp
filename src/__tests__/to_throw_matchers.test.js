/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 */

'use strict';

const jestExpect = require('../');

// Custom Error class because node versions have different stack trace strings.
class Error {
  constructor(message) {
    this.message = message;
    this.name = 'Error';
    this.stack =
      'Error\n' +
      '  at jestExpect' +
      ' (packages/expect/src/__tests__/toThrowMatchers-test.js:24:74)';
  }
}

['toThrowError', 'toThrow'].forEach(toThrow => {
  describe('.' + toThrow + '()', () => {
    class Err extends Error {}
    class Err2 extends Error {}

    test('to throw or not to throw', () => {
      jestExpect(() => {
        throw new Error('apple');
      })[toThrow]();
      jestExpect(() => {}).not[toThrow]();
    });

    describe('strings', () => {
      it('passes', () => {
        jestExpect(() => {
          throw new Error('apple');
        })[toThrow]('apple');
        jestExpect(() => {
          throw new Error('banana');
        }).not[toThrow]('apple');
        jestExpect(() => {}).not[toThrow]('apple');
      });

      test('did not throw at all', () => {
        expect(() =>
          jestExpect(() => {})[toThrow]('apple'),
        ).toThrowErrorMatchingSnapshot();
      });

      test('threw, but message did not match', () => {
        expect(() => {
          jestExpect(() => {
            throw new Error('apple');
          })[toThrow]('banana');
        }).toThrowErrorMatchingSnapshot();
      });

      it('properly escapes strings when matching against errors', () => {
        jestExpect(() => {
          throw new TypeError('"this"? throws.');
        })[toThrow]('"this"? throws.');
      });

      test('threw, but should not have', () => {
        expect(() => {
          jestExpect(() => {
            throw new Error('apple');
          }).not[toThrow]('apple');
        }).toThrowErrorMatchingSnapshot();
      });
    });

    describe('regexp', () => {
      it('passes', () => {
        expect(() => {
          throw new Error('apple');
        })[toThrow](/apple/);
        expect(() => {
          throw new Error('banana');
        }).not[toThrow](/apple/);
        expect(() => {}).not[toThrow](/apple/);
      });

      test('did not throw at all', () => {
        expect(() =>
          jestExpect(() => {})[toThrow](/apple/),
        ).toThrowErrorMatchingSnapshot();
      });

      test('threw, but message did not match', () => {
        expect(() => {
          jestExpect(() => {
            throw new Error('apple');
          })[toThrow](/banana/);
        }).toThrowErrorMatchingSnapshot();
      });

      test('threw, but should not have', () => {
        expect(() => {
          jestExpect(() => {
            throw new Error('apple');
          }).not[toThrow](/apple/);
        }).toThrowErrorMatchingSnapshot();
      });
    });

    describe('error class', () => {
      it('passes', () => {
        jestExpect(() => {
          throw new Err();
        })[toThrow](Err);
        jestExpect(() => {
          throw new Err();
        })[toThrow](Error);
        jestExpect(() => {
          throw new Err();
        }).not[toThrow](Err2);
        jestExpect(() => {}).not[toThrow](Err);
      });

      test('did not throw at all', () => {
        expect(() =>
          expect(() => {})[toThrow](Err),
        ).toThrowErrorMatchingSnapshot();
      });

      test('threw, but class did not match', () => {
        expect(() => {
          jestExpect(() => {
            throw new Err('apple');
          })[toThrow](Err2);
        }).toThrowErrorMatchingSnapshot();
      });

      test('threw, but should not have', () => {
        expect(() => {
          jestExpect(() => {
            throw new Err('apple');
          }).not[toThrow](Err);
        }).toThrowErrorMatchingSnapshot();
      });
    });

    test('invalid arguments', () => {
      expect(() =>
        jestExpect(() => {})[toThrow](111),
      ).toThrowErrorMatchingSnapshot();
    });

    test('invalid actual', () => {
      expect(() =>
        jestExpect('a string')[toThrow](),
      ).toThrowErrorMatchingSnapshot();
    });
  });
});
