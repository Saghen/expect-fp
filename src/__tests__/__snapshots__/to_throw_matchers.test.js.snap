// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toThrow() error class did not throw at all 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>type</><dim>)

Expected the function to throw an error of type:
  <green>\\"Err\\"</>
But it didn't throw anything."
`;

exports[`.toThrow() error class threw, but class did not match 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>type</><dim>)

Expected the function to throw an error of type:
  <green>\\"Err2\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrow() error class threw, but should not have 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrow(<green>type</><dim>)

Expected the function not to throw an error of type:
  <green>\\"Err\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrow() invalid actual 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>undefined</><dim>)

Received value must be a function, but instead \\"string\\" was found"
`;

exports[`.toThrow() invalid arguments 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrow(<green>number</><dim>)

Unexpected argument passed.
Expected: <green>\\"string\\"</>, <green>\\"Error (type)\\"</> or <green>\\"regexp\\"</>.
Got:
  string: <green>\\"111\\"</>"
`;

exports[`.toThrow() regexp did not throw at all 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>regexp</><dim>)

Expected the function to throw an error matching:
  <green>/apple/</>
But it didn't throw anything."
`;

exports[`.toThrow() regexp threw, but message did not match 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>regexp</><dim>)

Expected the function to throw an error matching:
  <green>/banana/</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrow() regexp threw, but should not have 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrow(<green>regexp</><dim>)

Expected the function not to throw an error matching:
  <green>/apple/</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrow() strings did not throw at all 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>string</><dim>)

Expected the function to throw an error matching:
  <green>\\"apple\\"</>
But it didn't throw anything."
`;

exports[`.toThrow() strings threw, but message did not match 1`] = `
"<dim>expect(<red>function</><dim>).toThrow(<green>string</><dim>)

Expected the function to throw an error matching:
  <green>\\"banana\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrow() strings threw, but should not have 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrow(<green>string</><dim>)

Expected the function not to throw an error matching:
  <green>\\"apple\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrowError() error class did not throw at all 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>type</><dim>)

Expected the function to throw an error of type:
  <green>\\"Err\\"</>
But it didn't throw anything."
`;

exports[`.toThrowError() error class threw, but class did not match 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>type</><dim>)

Expected the function to throw an error of type:
  <green>\\"Err2\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrowError() error class threw, but should not have 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrowError(<green>type</><dim>)

Expected the function not to throw an error of type:
  <green>\\"Err\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrowError() invalid actual 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>undefined</><dim>)

Received value must be a function, but instead \\"string\\" was found"
`;

exports[`.toThrowError() invalid arguments 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrowError(<green>number</><dim>)

Unexpected argument passed.
Expected: <green>\\"string\\"</>, <green>\\"Error (type)\\"</> or <green>\\"regexp\\"</>.
Got:
  string: <green>\\"111\\"</>"
`;

exports[`.toThrowError() regexp did not throw at all 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>regexp</><dim>)

Expected the function to throw an error matching:
  <green>/apple/</>
But it didn't throw anything."
`;

exports[`.toThrowError() regexp threw, but message did not match 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>regexp</><dim>)

Expected the function to throw an error matching:
  <green>/banana/</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrowError() regexp threw, but should not have 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrowError(<green>regexp</><dim>)

Expected the function not to throw an error matching:
  <green>/apple/</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrowError() strings did not throw at all 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>string</><dim>)

Expected the function to throw an error matching:
  <green>\\"apple\\"</>
But it didn't throw anything."
`;

exports[`.toThrowError() strings threw, but message did not match 1`] = `
"<dim>expect(<red>function</><dim>).toThrowError(<green>string</><dim>)

Expected the function to throw an error matching:
  <green>\\"banana\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;

exports[`.toThrowError() strings threw, but should not have 1`] = `
"<dim>expect(<red>function</><dim>).not.toThrowError(<green>string</><dim>)

Expected the function not to throw an error matching:
  <green>\\"apple\\"</>
Instead, it threw:
<red>  Error      </>
<red>      <dim>at jestExpect (packages/expect/src/__tests__/toThrowMatchers-test.js<dim>:24:74)</>"
`;
